src/compiler.ts

```diff
@@ -8578,18 +8589,38 @@ export class Compiler extends DiagnosticEmitter {
           return module.unreachable();
         }
 
-        let functionInstance = this.resolver.resolveFunction(
+        // Find the original function from our prototype
+        let functionInstanceCtxTypes = makeMap<string,Type>(flow.contextualTypeArguments);
+        let originalFunctionInstance = this.resolver.resolveFunction(
           functionPrototype,
           null,
-          makeMap<string,Type>(flow.contextualTypeArguments)
+          functionInstanceCtxTypes 
         );
-        if (!functionInstance || !this.compileFunction(functionInstance)) return module.unreachable();
+        if (!originalFunctionInstance || !this.compileFunction(originalFunctionInstance)) return module.unreachable();
+
+        // Create a new closure function for our original function
+        let closureFunctionInstance = new Function(
+          functionPrototype.name + "~anonymous|" + (actualFunction.nextAnonymousId++).toString(),
+          functionPrototype,
+          null,
+          originalFunctionInstance.signature.toClosureSignature(),
+          functionInstanceCtxTypes
+        );
+        if (!this.compileFunction(closureFunctionInstance)) return this.module.unreachable();
+
         if (contextualType.is(TypeFlags.HOST | TypeFlags.REFERENCE)) {
           this.currentType = Type.anyref;
-          return module.ref_func(functionInstance.internalName);
+          return module.ref_func(closureFunctionInstance.internalName);
         }
-        let index = this.ensureFunctionTableEntry(functionInstance);
-        this.currentType = functionInstance.signature.type;
+        console.log('torch2424 8615');
+
+        // Enter our closure function into our closures table
+        let index = this.ensureFunctionTableEntry(closureFunctionInstance);
+
+        // Set the param type of the function to the original function
+        this.currentType = originalFunctionInstance.signature.type;
+
+        // Return the closure function index
         return module.i32(index);
```

Comment:
We're still returning an index here, which is an invalid representation of a function. This has to be a pointer to a function struct like we do [here]( https://github.com/DuncanUszkay1/assemblyscript/blob/d69d861a8b7a4fe32e5f00908193e14c3541df4c/src/compiler.ts#L8335)

Reply:
Ah! Thanks for catching that! Cool I'll go ahead and fix that :smile: